### ISSUE TITLE
prefer-const false positive with destructured declaration

### ISSUE THREAD

**Issue by btmills** [eslint/eslint#8141]
**Tell us about your environment**

* **ESLint Version:** [ESLint demo](http://eslint.org/demo)
* **Node Version:** n/a
* **npm Version:** n/a

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

[ESLint demo](http://eslint.org/demo) with ECMA Version set to 6, all environments and all rules disabled.

**What did you do? Please include the actual source code causing the issue.**

```js
/* eslint prefer-const: "error" */
let [a, b] = foo();
b += 3;
```

**What did you expect to happen?**

No error. Because `b` is modified, in order to change `let` to `const`, I would instead have to change the code to

```js
/* eslint prefer-const: "error" */
const fooResult = foo();
const a = fooResult[0];
let b = fooResult[1];
b += 3;
```

**What actually happened? Please include the actual, raw output from ESLint.**

```
2:6 - 'a' is never reassigned. Use 'const' instead. (prefer-const)
```

**Comment by platinumazure** [comment]
Not to nitpick, but this doesn't really seem like a bug. The rule is correctly flagging `a` as something that could be rendered const. This seems more like a rule enhancement request to add an option to not warn in destructuring scenarios when at least one of the identifiers is reassigned.

Am I missing something?

**Comment by btmills** [comment]
> Am I missing something?

Nope, not at all! That's the exact counterargument I expected somebody to make. I don't have a strong opinion either way. I just tested and confirmed that it's not autofixed, so we're good there.

**Comment by gyandeeps** [comment]
@btmills I think this is how you want to configure the rule:

`/* eslint "prefer-const": [2, {"destructuring": "all"}] */`

**Comment by btmills** [comment]
@gyandeeps :flushed: you're exactly right - I should've read the docs to see if such an option already existed. Sorry for the noise, everyone!
